src/v/redpanda/application.cc

```diff
@@ -233,18 +233,20 @@ void application::configure_admin_server() {
                 .then([this, &server](
                         std::optional<ss::tls::credentials_builder> builder) {
                     if (!builder) {
-                        return;
+                        return ss::now();
                     }
-                    server.set_tls_credentials(
-                      builder
-                        ->build_reloadable_server_credentials(
-                          [this](
-                            const std::unordered_set<ss::sstring>& updated,
-                            const std::exception_ptr& eptr) {
-                              cluster::log_certificate_reload_event(
-                                _log, "API TLS", updated, eptr);
-                          })
-                        .get0());
```

Comment:
ah, crazy this went through.

Reply:
it looks like thread context assertion is only checked if the future isn't ready, so this seems plausible. that check should probably be made earlier in seastar.